lib/Doctrine/DBAL/Migrations/FileQueryWriter.php

```diff
@@ -4,51 +4,102 @@
 
 namespace Doctrine\DBAL\Migrations;
 
-/**
- * @since  1.6.0
- * @author Luís Cobucci <lcobucci@gmail.com>
- */
+use InvalidArgumentException;
+use function date;
+use function file_put_contents;
+use function is_dir;
+use function sprintf;
+
 final class FileQueryWriter implements QueryWriter
 {
-    /**
-     * @var string
-     */
+    /** @var string */
     private $columnName;
 
-    /**
-     * @var string
-     */
+    /** @var string */
     private $tableName;
 
-    /**
-     * @var null|OutputWriter
-     */
+    /** @var null|OutputWriter */
     private $outputWriter;
 
-    public function __construct(string $columnName, string $tableName, ?OutputWriter $outputWriter)
-    {
+    public function __construct(
+        string $columnName,
+        string $tableName,
+        ?OutputWriter $outputWriter
+    ) {
         $this->columnName   = $columnName;
         $this->tableName    = $tableName;
         $this->outputWriter = $outputWriter;
     }
 
     /**
-     * TODO: move SqlFileWriter's behaviour to this class - and kill it with fire (on the next major release)
-     * @param string $path
-     * @param string $direction
-     * @param array $queriesByVersion
-     * @return bool
+     * @param mixed[] $queriesByVersion
      */
-    public function write(string $path, string $direction, array $queriesByVersion) : bool
+    public function write(
+        string $path,
+        string $direction,
+        array $queriesByVersion
+    ) : bool {
+        $path   = $this->buildMigrationFilePath($path);
+        $string = $this->buildMigrationFile($queriesByVersion, $direction);
+
+        if ($this->outputWriter) {
+            $this->outputWriter->write(
+                "\n" . sprintf('Writing migration file to "<info>%s</info>"', $path)
+            );
+        }
+
+        return file_put_contents($path, $string) ? true : false;
```

Comment:
should be `file_put_contents(...) !== false` and without ternary, if this stands for error check